src/client/common/variables/environmentVariablesProvider.ts

```diff
@@ -104,6 +87,23 @@ export class EnvironmentVariablesProvider implements IEnvironmentVariablesProvid
             this.disposables.push(envFileWatcher.onDidDelete(() => this.onEnvironmentFileChanged(workspaceFolderUri)));
         }
     }
+    private async _getEnvironmentVariables(resource?: Uri): Promise<EnvironmentVariables> {
+        let customEnvVars = await this.getCustomEnvironmentVariables(resource);
+        if (customEnvVars) {
+            customEnvVars = {};
+        }
+        const mergedVars: EnvironmentVariables = {};
+        this.envVarsService.mergeVariables(process.env, mergedVars); // Copy current proc vars into new obj.
+        this.envVarsService.mergeVariables(customEnvVars!, mergedVars); // Copy custom vars over into obj.
+        const pathVariable = this.platformService.pathVariableName;
+        if (customEnvVars![pathVariable]) {
+            this.envVarsService.appendPath(mergedVars!, customEnvVars![pathVariable]!);
```

Comment:
Does this function work the opposiste of mergeVariables? The destination is the first argument?

Reply:
Yes
The source is a string.

Reply:
Do you want me to flip it?

Reply:
Maybe. Or do like Ian suggested for the other function (it returns the appended vars instead of modifying the input param)

Reply:
But after your last fix, I'm okay with how it is too.

Reply:
Looks good. I think that you could remove the !s now that are on the customEnvVars uses. Looks like it should never be undefined now.